### PR TITLE
feat: add the transfer fees and spend queue logic

### LINKED ISSUE #75: Pull in transfer priority queue from old repo

### PR DESCRIPTION
**Pull request by oetyng**
Resolves #75.

### CODE REVIEW

File: safenode/src/protocol/types/fees/required_fee_content.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+use serde::{Deserialize, Serialize};
+use tiny_keccak::{Hasher, Sha3};
+
+/// The content of a required fee.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct RequiredFeeContent {
+    /// A simple u64, encrypted to the `DbcId` of the `Dbc` being spent,
+    /// for which a fee is being required.
+    pub amount_cipher: Ciphertext,
```
Comment (grumbach) [inline safenode/src/protocol/types/fees/required_fee_content.rs:21]: Does that mean fees are for each DBCs??

File: safenode/src/protocol/types/fees/spend_queue.rs [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+        match priority {
+            SpendPriority::Highest => self.highest,
+            SpendPriority::High => self.high,
+            SpendPriority::MediumHigh => self.medium_high,
+            SpendPriority::Normal => self.avg,
+            SpendPriority::MediumLow => self.medium_low,
+            SpendPriority::Low => self.low,
+            SpendPriority::Lowest => self.lowest,
+        }
+    }
```
Comment (grumbach) [inline safenode/src/protocol/types/fees/spend_queue.rs:120]: Why do we need this?

**Review by grumbach — COMMENTED** [review]
What is `SpendPriority` and why do we need it?

File: safenode/src/protocol/types/fees/required_fee_content.rs (hunk 1 above)
Comment (oetyng) [reply safenode/src/protocol/types/fees/required_fee_content.rs:21]: For each input dbc to spend, yes.

**Comment by oetyng** [comment]
> What is `SpendPriority` and why do we need it?

https://safenetforum.org/t/update-30-march-2023/38175#spend-priority-queue-8

**Review by dirvine — APPROVED** [review]

**Pull request merged**